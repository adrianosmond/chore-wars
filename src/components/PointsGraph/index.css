.points-graph {
  position: sticky;
  z-index: 2;
  top: calc(-6rem - ((100vw - 4rem)/2));
  background-color: #fff;
  padding: 1rem 1rem 2rem;
  margin-top: -1rem;
  margin-left: -1rem;
  margin-right: -1rem;
}

/* After 464px the width of the avatars won't grow any more
   so the top coordinate of the bar becomes constant */
@media screen and (min-width: 432px) {
  .points-graph {
    top: calc(-5rem - 200px);
  }
}

.points-graph__title {
  margin: 0;
  line-height: 1;
  text-align: center;
}

.points-graph__people {
  display: flex;
  justify-content: space-around;
  width: 100%;
}

.points-graph__person {
  width: 200px;
}

.points-graph__person-picture {
  display: block;
}

.points-graph__person-name {
  font-weight: bold;
  line-height: 1;
  text-align: center;
  margin-top: 1rem;
}

.points-graph__graph {
  margin-top: 2rem;
  margin-bottom: 2rem;
  position: relative;
}

.points-graph__graph::before {
  content: "";
  position: absolute;
  height: 0.25rem;
  width: 100%;
  background-color: #ddd;
}

.points-graph__bar {
  position: absolute;
  right: 50%;
  height: 0.25rem;
  background-color: tomato;
  transition: width 1s;
}

.points-graph__bar--2 {
  left: 50%;
  right: auto;
}

.points-graph__difference {
  position: absolute;
  bottom: 0.5rem;
  padding: 0.25rem 0.375rem;
  font-size: 0.75rem;
  border-radius: 0.25rem;
  background-color: tomato;
  color: #fff;
}

.points-graph__difference::after {
  content: "";
  position: absolute;
  bottom: -3px;
  left: 50%;
  margin-left: -3px;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 3px 3px 0 3px;
  border-color: tomato transparent transparent transparent;
}

.points-graph__difference--1 {
  transform: translateX(-50%);
  left: 1px;
}

.points-graph__difference--2 {
  transform: translateX(50%);
  right: 1px;
}

@media screen and (min-width: 37.5em) {
  .points-graph {
  /* In the 2 col layout, the width of the avatars grows
     so the top coordinate of the bar becomes thicker */
    top: calc(-6.5rem - 200px);
    padding-top: 2rem;
  }

  .points-graph__person-name {
    font-size: 1.5rem;
  }

  .points-graph__difference {
    font-size: 1rem;
  }
}
