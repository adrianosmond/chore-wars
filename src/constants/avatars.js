export const avatarOptions = {
  skinColor: [
    'Pale',
    'Light',
    'Tanned',
    'Yellow',
    'Brown',
    'DarkBrown',
    'Black',
  ],
  topType: [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairBigHair',
    'LongHairBob',
    'LongHairBun',
    'LongHairCurly',
    'LongHairCurvy',
    'LongHairDreads',
    'LongHairFrida',
    'LongHairFro',
    'LongHairFroBand',
    'LongHairNotTooLong',
    'LongHairShavedSides',
    'LongHairMiaWallace',
    'LongHairStraight',
    'LongHairStraight2',
    'LongHairStraightStrand',
    'ShortHairDreads01',
    'ShortHairDreads02',
    'ShortHairFrizzle',
    'ShortHairShaggyMullet',
    'ShortHairShortCurly',
    'ShortHairShortFlat',
    'ShortHairShortRound',
    'ShortHairShortWaved',
    'ShortHairSides',
    'ShortHairTheCaesar',
    'ShortHairTheCaesarSidePart',
  ],
  hatColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
  hairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'PastelPink',
    'Platinum',
    'Red',
    'SilverGray',
  ],
  facialHairType: [
    'Blank',
    'BeardMedium',
    'BeardLight',
    'BeardMagestic',
    'MoustacheFancy',
    'MoustacheMagnum',
  ],
  facialHairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'Platinum',
    'Red',
  ],
  accessoriesType: [
    'Blank',
    'Kurt',
    'Prescription01',
    'Prescription02',
    'Round',
    'Sunglasses',
    'Wayfarers',
  ],
  clotheType: [
    'BlazerShirt',
    'BlazerSweater',
    'CollarSweater',
    'Hoodie',
    'Overall',
    'ShirtCrewNeck',
    'ShirtScoopNeck',
    'ShirtVNeck',
  ],
  clotheColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
};


export const editorOrder = [
  'skinColor',
  'topType',
  'hairColor',
  'hatColor',
  'facialHairType',
  'facialHairColor',
  'accessoriesType',
  'clotheType',
  'clotheColor',
];

export const labels = {
  skinColor: 'Skin Colour',
  topType: 'Hair or Headware',
  hairColor: 'Hair Colour',
  hatColor: 'Headware Colour',
  facialHairType: 'Facial Hair Type',
  facialHairColor: 'Facial Hair Colour',
  accessoriesType: 'Glasses',
  clotheType: 'Clothing Type',
  clotheColor: 'Clothing Colour',
};

export const isEditable = {
  skinColor: () => true,
  topType: () => true,
  hairColor: options => [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairFrida',
    'LongHairShavedSides',
  ].indexOf(options.topType) < 0,
  hatColor: options => [
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
  ].indexOf(options.topType) >= 0,
  facialHairType: options => options.topType !== 'Hijab',
  facialHairColor: options => isEditable.facialHairType(options) && options.facialHairType !== 'Blank',
  accessoriesType: () => true,
  clotheType: () => true,
  clotheColor: options => !options.clotheType.startsWith('Blazer'),
};

export const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};
