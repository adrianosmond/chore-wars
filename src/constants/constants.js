const TIME_UNIT = 1000 * 60 * 60 * 24; // 1 day
const MIN_CHORE_FREQUENCY = 1;
const MAX_CHORE_FREQUENCY = 365;
const MIN_CHORE_POINTS = 1;
const MAX_CHORE_POINTS = 200;

const DATE_FORMAT = 'DD/MM/YYYY';

const MAX_POINT_DIFFERENCE = 500;

const MAX_NAME_LENGTH = 10;
const JOIN_CODE_LENGTH = 8;

const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};

const ActionTypes = {
  addChore: 'ADD_CHORE',
  resetChoreDoneDate: 'RESET_CHORE_DONE_DATE',
  blockChore: 'BLOCK_CHORE',
  unblockChore: 'UNBLOCK_CHORE',
  removeChore: 'REMOVE_CHORE',
  updateChore: 'UPDATE_CHORE',
  setChores: 'SET_CHORES',
  makeChain: 'MAKE_CHAIN',
  breakChain: 'BREAK_CHAIN',
  addPoints: 'ADD_POINTS',
  claimPrize: 'CLAIM_PRIZE',
  paidDebt: 'PAID_DEBT',
  setPoints: 'SET_POINTS',
  updateUser: 'UPDATE_USER',
  setAuthUser: 'SET_AUTH_USER',
  setGame: 'SET_GAME',
  setPointsLoaded: 'SET_POINTS_LOADED',
  setChoresLoaded: 'SET_CHORES_LOADED',
  saveStatePostUndo: 'SAVE_STATE_POST_UNDO',
};

export {
  TIME_UNIT,
  MIN_CHORE_FREQUENCY,
  MAX_CHORE_FREQUENCY,
  MIN_CHORE_POINTS,
  MAX_CHORE_POINTS,
  MAX_NAME_LENGTH,
  JOIN_CODE_LENGTH,
  DATE_FORMAT,
  MAX_POINT_DIFFERENCE,
  ActionTypes,
  DefaultAvatar,
};
