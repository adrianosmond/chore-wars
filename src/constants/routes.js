export const LOGIN = '/'
export const CHORES = '/chores'
export const NEW_CHORE = '/chores/new'
