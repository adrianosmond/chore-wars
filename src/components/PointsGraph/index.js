import React from 'react';
import { connect } from 'react-redux';
import { MAX_POINT_DIFFERENCE } from '../../constants/constants';
import './index.css';

const PointsGraph = (props) => {
  const { points } = props;
  const players = Object.keys(props.points);
  const minPoints = Math.min(points[players[0]].points, points[players[1]].points);
  const scoresTied = points[players[0]].points === points[players[1]].points;
  return (
    <div className="points-graph">
      <h1 className="points-graph__title">Chore Wars</h1>
      <div className="points-graph__people">
        {players.map((playerId, idx) => {
          const player = points[playerId];
          return (
            <div className={`points-graph__person points-graph__person--${idx + 1}`} key={playerId}>
              <img src={`/images/avatars/${playerId}.svg`} className="points-graph__person-picture" alt={''} />
              <div className="points-graph__person-name">{player.name}</div>
            </div>
          );
        })}
      </div>
      <div className="points-graph__graph">
        {players.map((playerId, idx) => {
          const player = points[playerId];
          const playerBar = (player.points - minPoints);
          return (
            <div className={`points-graph__bar points-graph__bar--${idx + 1}`} style={{ width: `${Math.min(50, (playerBar / MAX_POINT_DIFFERENCE) * 50)}%` }} key={playerId}>
              { playerBar || scoresTied ? <div className={`points-graph__difference points-graph__difference--${idx + 1}`}>{playerBar}</div> : null }
            </div>
          );
        })}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  points: state.points.present,
});

export { PointsGraph };

export default connect(mapStateToProps)(PointsGraph);
