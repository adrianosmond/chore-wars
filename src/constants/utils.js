import { TIME_UNIT } from './constants'
const sortByCurrentPoints = (a, b) => b.currentPoints - a.currentPoints

const processChore = (chore) => {
  const timeSinceChore = (new Date().getTime() - chore.lastDone) / TIME_UNIT
  const timeRemaining = timeSinceChore - chore.frequency
  const percentage = Math.min(100 * timeSinceChore / chore.frequency, 100)
  const multiplier = 1 + (1 / chore.frequency)
  const due = chore.lastDone + (chore.frequency * TIME_UNIT)
  return {
    currentPoints: Math.round(percentage / 100 * chore.pointsPerTime * Math.pow(multiplier, timeRemaining)),
    percentage,
    due
  }
}

export { sortByCurrentPoints, processChore }
