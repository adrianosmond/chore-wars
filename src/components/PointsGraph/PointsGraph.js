import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Avatar from 'avataaars';
import Confetti from 'react-dom-confetti';

import PopUpMenu from '../PopUpMenu';

import { claimPrize, paidDebt } from '../../actions/pointActions';

import { MAX_POINT_DIFFERENCE } from '../../constants/constants';
import * as routes from '../../constants/routes';

import './PointsGraph.css';

const PointsGraph = (props) => {
  const { points } = props;
  const players = Object.keys(props.points);

  const minPoints = players.length === 2 ?
    Math.min(points[players[0]].points, points[players[1]].points) :
    points[players[0]].points;

  const scoresTied = players.length === 2 ?
    points[players[0]].points === points[players[1]].points :
    true;

  return (
    <div className="points-graph">
      <h1 className="points-graph__title">Chore Wars</h1>
      <div className="points-graph__people">
        {players.map((playerId, idx) => {
          const player = points[playerId];
          const playerPicture = (
            <div className="points-graph__person-picture">
              <Avatar style={{ width: '100%', height: '100%', display: 'block' }} { ...player.avatar } />
              { player.isOwed > 0 ?
              <PopUpMenu extraClasses={`points-graph__paid-button points-graph__paid-button--${idx + 1}`}
                side={ idx === 0 ? 'left' : 'right' }
                options={[{
                  type: 'button',
                  text: `${player.name} was paid back`,
                  onClick: () => props.paidDebt(playerId, props.gameId),
                }]}>
                <span className="points-graph__person-owed" role="img" aria-labelledby={`${player.name} is owed`}>ðŸŒŸ</span>
                { player.isOwed > 1 ? <span className="points-graph__owed-multiple">{player.isOwed}</span> : null }
              </PopUpMenu>
              : null}
            </div>
          );
          const playerName = <div className="points-graph__person-name">{player.name}</div>;
          if (playerId === props.user) {
            return (
              <Link to={routes.PROFILE_EDITOR} className={`points-graph__person points-graph__person--${idx + 1}`} key={playerId}>
                {playerPicture}
                {playerName}
              </Link>
            );
          }
          return (
            <div className={`points-graph__person points-graph__person--${idx + 1}`} key={playerId}>
              {playerPicture}
              {playerName}
            </div>
          );
        })}
        {players.length === 2 ? null :
        <div className="points-graph__person">
          <p>Invite someone to sign up and give them the code:</p>
          <p>{points[players[0]].joinCode}</p>
          <p>to get them to join this game.</p>
        </div>}
      </div>
      <div className="points-graph__graph">
        {players.map((playerId, idx) => {
          const player = points[playerId];
          const playerBar = (player.points - minPoints);
          const winning = playerBar >= MAX_POINT_DIFFERENCE;
          return (
            <div className={`points-graph__bar points-graph__bar--${idx + 1}${winning ? ' points-graph__bar--winning' : ''}`}
              style={{ width: `${Math.min(50, (playerBar / MAX_POINT_DIFFERENCE) * 50)}%` }}
              key={playerId}>
              { playerBar || scoresTied ?
              <div className={`points-graph__difference points-graph__difference--${idx + 1}${winning ? ' points-graph__difference--winning' : ''}`}
                onClick={winning ? () => props.claimPrize(playerId, props.gameId) : null}>
                <Confetti active={!winning} />
                {playerBar}
                </div>
              : null }
            </div>
          );
        })}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  points: state.points.present,
  user: state.session.authUser.uid,
  gameId: state.session.game.gameId,
});

const matchDispatchToProps = dispatch => ({
  claimPrize: (player, game) => dispatch(claimPrize(player, game)),
  paidDebt: (player, game) => dispatch(paidDebt(player, game)),
});

export { PointsGraph };

export default connect(mapStateToProps, matchDispatchToProps)(PointsGraph);
