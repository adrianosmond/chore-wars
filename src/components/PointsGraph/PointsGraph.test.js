import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { MAX_POINT_DIFFERENCE } from 'constants/constants';
import PointsGraph from './PointsGraph';

const mockStore = configureMockStore([thunk]);

describe('PointsGraph', () => {
  it('Renders an equal points graph', () => {
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: 100,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: 100,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with player 1 winning', () => {
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: 200,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: 100,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with player 2 winning', () => {
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: 100,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: 200,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with a difference of > MAX with p1 winning', () => {
    const p2Score = 100;
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: p2Score + MAX_POINT_DIFFERENCE + 100,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: p2Score,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with a difference of > MAX with p2 winning', () => {
    const p1Score = 100;
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: p1Score,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: p1Score + MAX_POINT_DIFFERENCE + 100,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with p1 being owed', () => {
    const p2Score = 100;
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: p2Score + MAX_POINT_DIFFERENCE + 100,
            isOwed: 1,
          },
          player2: {
            name: 'Player 2',
            points: p2Score,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with p2 being owed', () => {
    const p1Score = 100;
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: p1Score,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: p1Score + MAX_POINT_DIFFERENCE + 100,
            isOwed: 1,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with p1 being owed > 1', () => {
    const p2Score = 100;
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: p2Score + MAX_POINT_DIFFERENCE + 100,
            isOwed: 2,
          },
          player2: {
            name: 'Player 2',
            points: p2Score,
            isOwed: 0,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });

  it('Renders a points graph with p2 being owed > 1', () => {
    const p1Score = 100;
    const store = mockStore({
      points: {
        present: {
          player1: {
            name: 'Player 1',
            points: p1Score,
            isOwed: 0,
          },
          player2: {
            name: 'Player 2',
            points: p1Score + MAX_POINT_DIFFERENCE + 100,
            isOwed: 2,
          },
        },
      },
      session: {
        authUser: 'player1',
        game: {
          gameId: 'test-game',
        },
      },
    });
    expect(shallow(<PointsGraph store={store} />).dive()).toMatchSnapshot();
  });
});
