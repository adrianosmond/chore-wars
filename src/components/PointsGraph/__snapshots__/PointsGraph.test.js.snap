// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsGraph Renders a points graph with a difference of > MAX with p1 winning 1`] = `
<div
  className="points-graph"
>
  <h1
    className="points-graph__title"
  >
    Chore Wars
  </h1>
  <div
    className="points-graph__people"
  >
    <div
      className="points-graph__person points-graph__person--1"
      key="player1"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 1
      </div>
    </div>
    <div
      className="points-graph__person points-graph__person--2"
      key="player2"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 2
      </div>
    </div>
  </div>
  <div
    className="points-graph__graph"
  >
    <div
      className="points-graph__bar points-graph__bar--1"
      key="player1"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <div
        className="points-graph__difference points-graph__difference--1"
      >
        600
      </div>
    </div>
    <div
      className="points-graph__bar points-graph__bar--2"
      key="player2"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;

exports[`PointsGraph Renders a points graph with a difference of > MAX with p2 winning 1`] = `
<div
  className="points-graph"
>
  <h1
    className="points-graph__title"
  >
    Chore Wars
  </h1>
  <div
    className="points-graph__people"
  >
    <div
      className="points-graph__person points-graph__person--1"
      key="player1"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 1
      </div>
    </div>
    <div
      className="points-graph__person points-graph__person--2"
      key="player2"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 2
      </div>
    </div>
  </div>
  <div
    className="points-graph__graph"
  >
    <div
      className="points-graph__bar points-graph__bar--1"
      key="player1"
      style={
        Object {
          "width": "0%",
        }
      }
    />
    <div
      className="points-graph__bar points-graph__bar--2"
      key="player2"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <div
        className="points-graph__difference points-graph__difference--2"
      >
        600
      </div>
    </div>
  </div>
</div>
`;

exports[`PointsGraph Renders a points graph with player 1 winning 1`] = `
<div
  className="points-graph"
>
  <h1
    className="points-graph__title"
  >
    Chore Wars
  </h1>
  <div
    className="points-graph__people"
  >
    <div
      className="points-graph__person points-graph__person--1"
      key="player1"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 1
      </div>
    </div>
    <div
      className="points-graph__person points-graph__person--2"
      key="player2"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 2
      </div>
    </div>
  </div>
  <div
    className="points-graph__graph"
  >
    <div
      className="points-graph__bar points-graph__bar--1"
      key="player1"
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <div
        className="points-graph__difference points-graph__difference--1"
      >
        100
      </div>
    </div>
    <div
      className="points-graph__bar points-graph__bar--2"
      key="player2"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;

exports[`PointsGraph Renders a points graph with player 2 winning 1`] = `
<div
  className="points-graph"
>
  <h1
    className="points-graph__title"
  >
    Chore Wars
  </h1>
  <div
    className="points-graph__people"
  >
    <div
      className="points-graph__person points-graph__person--1"
      key="player1"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 1
      </div>
    </div>
    <div
      className="points-graph__person points-graph__person--2"
      key="player2"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 2
      </div>
    </div>
  </div>
  <div
    className="points-graph__graph"
  >
    <div
      className="points-graph__bar points-graph__bar--1"
      key="player1"
      style={
        Object {
          "width": "0%",
        }
      }
    />
    <div
      className="points-graph__bar points-graph__bar--2"
      key="player2"
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <div
        className="points-graph__difference points-graph__difference--2"
      >
        100
      </div>
    </div>
  </div>
</div>
`;

exports[`PointsGraph Renders an equal points graph 1`] = `
<div
  className="points-graph"
>
  <h1
    className="points-graph__title"
  >
    Chore Wars
  </h1>
  <div
    className="points-graph__people"
  >
    <div
      className="points-graph__person points-graph__person--1"
      key="player1"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 1
      </div>
    </div>
    <div
      className="points-graph__person points-graph__person--2"
      key="player2"
    >
      <div
        className="points-graph__person-picture"
      >
        <AvatarComponent
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="points-graph__person-name"
      >
        Player 2
      </div>
    </div>
  </div>
  <div
    className="points-graph__graph"
  >
    <div
      className="points-graph__bar points-graph__bar--1"
      key="player1"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      <div
        className="points-graph__difference points-graph__difference--1"
      >
        0
      </div>
    </div>
    <div
      className="points-graph__bar points-graph__bar--2"
      key="player2"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      <div
        className="points-graph__difference points-graph__difference--2"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
